crates/rethnet_evm/benches/state.rs

```diff
@@ -0,0 +1,133 @@
+use criterion::{criterion_group, criterion_main, BatchSize, BenchmarkId, Criterion};
+use revm::{db::StateRef, primitives::AccountInfo};
+use std::{clone::Clone, str::FromStr};
+
+use rethnet_eth::Address;
+use rethnet_evm::state::{HybridState, LayeredState, RethnetLayer, StateError, SyncState};
+
+#[derive(Default)]
+struct RethnetStates {
+    layered: LayeredState<RethnetLayer>,
+    hybrid: HybridState<RethnetLayer>,
+}
+
+impl RethnetStates {
+    fn fill(&mut self, number_of_accounts: u64, number_of_accounts_per_checkpoint: u64) {
+        let mut states: [&mut dyn SyncState<StateError>; 2] = [&mut self.layered, &mut self.hybrid];
+        for state in states.iter_mut() {
+            let number_of_checkpoints = number_of_accounts / number_of_accounts_per_checkpoint;
+            for _ in 0..=number_of_checkpoints {
+                for i in 1..=number_of_accounts {
+                    state
+                        .insert_account(
+                            Address::from_str(&format!("0x{:0>40x}", i)).unwrap(),
```

Comment:
One of the `from_low_u64_*` methods will likely be a lot faster than creating and parsing a string.

Reply:
I quickly whipped up a couple of benchmarks to confirm your hunch, and you're right:
![image](https://user-images.githubusercontent.com/7883777/234660110-c828bc74-650a-40d6-969e-c1b5b4f2b067.png)
Addressed in https://github.com/NomicFoundation/hardhat/pull/3868/commits/9e7c6bed702a8d753e487e953197b2e17336bfbc